processor/src/main/java/com/linecorp/decaton/processor/runtime/ProcessorProperties.java

```diff
@@ -81,6 +81,18 @@ public class ProcessorProperties extends AbstractDecatonProperties {
     public static final PropertyDefinition<Integer> CONFIG_PARTITION_CONCURRENCY =
             PropertyDefinition.define("decaton.partition.concurrency", Integer.class, 1,
                                       v -> v instanceof Integer && (Integer) v > 0);
+
+    /**
+     * Whether to guarantee ordering of tasks by key.
+     * if true, tasks with same key will be processed in a same partition.
+     * if false, tasks with same key may be processed in different partitions.
```

Comment:
Add a comment explaining why processor tasks with same key may be processed in different partitions if CONFIG_PARTITION_BY_KEY is set to false.